src/components/application_manager/test/commands/mobile/on_button_notification_commands_test .cc

```diff
@@ -0,0 +1,331 @@
+      .WillOnce(Return(empty_subscribed_apps_list));
+
+  EXPECT_CALL(this->app_mngr_, SendMessageToMobile(_, _)).Times(0);
+
+  command->Run();
+}
+
+TYPED_TEST(OnButtonNotificationCommandsTest, Run_InvalidHmiLevel_UNSUCCESS) {
+  typedef typename TestFixture::Notification Notification;
+
+  MessageSharedPtr notification_msg(
+      this->CreateMessage(smart_objects::SmartType_Map));
+
+  (*notification_msg)[am::strings::msg_params][am::hmi_response::button_name] =
+      kButtonName;
+
+  SharedPtr<Notification> command(
+      this->template CreateCommand<Notification>(notification_msg));
+
+  typename TestFixture::MockAppPtr mock_app = this->CreateMockApp();
+  std::vector<ApplicationSharedPtr> subscribed_apps_list;
+  subscribed_apps_list.push_back(mock_app);
+
+  EXPECT_CALL(*mock_app, hmi_level())
+      .WillRepeatedly(Return(mobile_apis::HMILevel::HMI_NONE));
+
+  EXPECT_CALL(this->app_mngr_, applications_by_button(kButtonName))
+      .WillOnce(Return(subscribed_apps_list));
+
+  EXPECT_CALL(this->app_mngr_, SendMessageToMobile(_, _)).Times(0);
+
+  command->Run();
+}
+
+TYPED_TEST(OnButtonNotificationCommandsTest,
+           Run_ButtonOkOnlyForHmiLevelFull_UNSUCCESS) {
+  typedef typename TestFixture::Notification Notification;
+
+  MessageSharedPtr notification_msg(
+      this->CreateMessage(smart_objects::SmartType_Map));
+
+  (*notification_msg)[am::strings::msg_params][am::hmi_response::button_name] =
+      kButtonName;
+
+  SharedPtr<Notification> command(
+      this->template CreateCommand<Notification>(notification_msg));
+
+  typename TestFixture::MockAppPtr mock_app = this->CreateMockApp();
+  std::vector<ApplicationSharedPtr> subscribed_apps_list;
+  subscribed_apps_list.push_back(mock_app);
+
+  EXPECT_CALL(*mock_app, hmi_level())
+      .WillRepeatedly(Return(mobile_apis::HMILevel::HMI_LIMITED));
+
+  EXPECT_CALL(this->app_mngr_, applications_by_button(kButtonName))
+      .WillOnce(Return(subscribed_apps_list));
+
+  EXPECT_CALL(this->app_mngr_, SendMessageToMobile(_, _)).Times(0);
+
+  command->Run();
+}
+
+TYPED_TEST(OnButtonNotificationCommandsTest, Run_SUCCESS) {
+  typedef typename TestFixture::Notification Notification;
+
+  MessageSharedPtr notification_msg(
+      this->CreateMessage(smart_objects::SmartType_Map));
+
+  (*notification_msg)[am::strings::msg_params][am::hmi_response::button_name] =
+      kButtonName;
+
+  SharedPtr<Notification> command(
+      this->template CreateCommand<Notification>(notification_msg));
+
+  typename TestFixture::MockAppPtr mock_app = this->CreateMockApp();
+  std::vector<ApplicationSharedPtr> subscribed_apps_list;
+  subscribed_apps_list.push_back(mock_app);
+
+  EXPECT_CALL(*mock_app, hmi_level())
+      .WillRepeatedly(Return(mobile_apis::HMILevel::HMI_FULL));
+
+  EXPECT_CALL(this->app_mngr_, applications_by_button(kButtonName))
+      .WillOnce(Return(subscribed_apps_list));
+
+  EXPECT_CALL(this->app_mngr_,
```

Comment:
@LevchenkoS I think you should create article on confluence with description why we should use ```this``` for some method calls and member usage. It will not very obvious for new people who will write test for commands.